src/core/ActivityInfoTree.cpp

```diff
@@ -139,16 +139,35 @@ void ActivityInfoTree::filterByTag(const QString &tag)
     for(auto activity: m_menuTreeFull) {
         if((activity->section().indexOf(tag) != -1 ||
 			tag == "all" ||
-            (tag == "favorite" && activity->favorite())) &&
-            (activity->difficulty() >= ApplicationSettings::getInstance()->filterLevelMin() &&
-             activity->difficulty() <= ApplicationSettings::getInstance()->filterLevelMax())) {
-            m_menuTree.push_back(activity);
+            (tag == "favorite" && activity->favorite()))) {
+            filterByDifficulty(activity,ApplicationSettings::getInstance()->filterLevelMin(),
```

Comment:
I wouldn't filter by difficulty activity by activity but only once at the end of the for loop